service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -0,0 +1,232 @@
+        } else {
+            hasHearings = NO.getValue();
+        }
+
+        ManageDocument manageDocument = ManageDocument.builder().hasHearings(hasHearings).build();
+
+        if (caseData.hasC2DocumentBundle()) {
+            listAndLabel.put(SUPPORTING_C2_LIST_KEY, caseData.buildC2DocumentDynamicList());
+        }
+
+        listAndLabel.put(MANAGE_DOCUMENT_KEY, manageDocument);
+
+        return listAndLabel;
+    }
+
+    public Map<String, Object> initialiseHearingListAndLabel(CaseData caseData) {
+        Map<String, Object> listAndLabel = new HashMap<>();
+
+        if (caseData.getManageDocument().isDocumentRelatedToHearing()) {
+            UUID selectedHearingCode = getDynamicListValueCode(caseData.getManageDocumentsHearingList(), mapper);
+            HearingBooking hearingBooking = getHearingBookingByUUID(caseData.getHearingDetails(), selectedHearingCode);
+
+            listAndLabel.put(MANAGE_DOCUMENTS_HEARING_LABEL_KEY, hearingBooking.toLabel(DATE));
+            listAndLabel.put(MANAGE_DOCUMENTS_HEARING_LIST_KEY, caseData.buildDynamicHearingList(selectedHearingCode));
+        }
+
+        return listAndLabel;
+    }
+
+    public Map<String, Object> initialiseC2DocumentListAndLabel(CaseData caseData) {
+        Map<String, Object> listAndLabel = new HashMap<>();
+
+        UUID selectedC2DocumentCode = getDynamicListValueCode(caseData.getManageDocumentsSupportingC2List(), mapper);
+        List<Element<C2DocumentBundle>> c2DocumentBundle = caseData.getC2DocumentBundle();
+
+        IntStream.range(0, c2DocumentBundle.size())
+            .filter(index -> c2DocumentBundle.get(index).getId().equals(selectedC2DocumentCode))
+            .findFirst()
+            .ifPresent(index -> listAndLabel.put(SUPPORTING_C2_LABEL,
+                c2DocumentBundle.get(index).getValue().toLabel(index + 1)));
+
+        listAndLabel.put(SUPPORTING_C2_LIST_KEY, caseData.buildC2DocumentDynamicList(selectedC2DocumentCode));
+
+        return listAndLabel;
+    }
+
+    public List<Element<SupportingEvidenceBundle>> getFurtherEvidenceCollection(CaseData caseData) {
+        if (caseData.getManageDocument().isDocumentRelatedToHearing()) {
+            List<Element<HearingFurtherEvidenceBundle>> bundles = caseData.getHearingFurtherEvidenceDocuments();
+            if (!bundles.isEmpty()) {
+                UUID selectedHearingCode = getDynamicListValueCode(caseData.getManageDocumentsHearingList(), mapper);
+
+                Optional<Element<HearingFurtherEvidenceBundle>> bundle = findElement(
+                    selectedHearingCode, bundles
+                );
+
+                if (bundle.isPresent()) {
+                    return bundle.get().getValue().getSupportingEvidenceBundle();
+                }
+            }
+        } else if (caseData.getFurtherEvidenceDocuments() != null) {
+            return caseData.getFurtherEvidenceDocuments();
+        }
+
+        return getSupportingEvidenceBundle(null);
```

Comment:
I don't like the usage of this method here, we are invoking where we know it's just going to return as empty element. Couldn't we have different method? It seems confusing get with passing null, you need to jump in to see what's happening there.